WordPress/gutenberg
# [`@wordpress/notices`] Add a new action `removeNotices` which allows bulk removal of notices

## What?

Adds a new action on the `core/notices` store that will remove several notices at once by supplying multiple IDs.

`removeNotices( [ 'some-id', 'some-other-id' ] )`

## Why?

To remove multiple notices, you'd need to make several calls to `removeNotice` this makes a slight performance improvement by only dispatching one action.

An example use case is in WooCommerce Blocks where we need to [remove several notices of a certain type](https://github.com/woocommerce/woocommerce-blocks/blob/fbd412642e828e48e3c4c19b391a290afbacccd4/assets/js/utils/notices.ts#L15).

Currently we loop through each notice and dispatch the `removeNotice` action for each of them. It would be more performant if we could dispatch a single action and achieve the same result.

## How?

Adds a new action creator and the action is handled in a new case in the reducer.

## Testing Instructions




1. Run the automated tests.
2. Run the following code in your console:
```js
wp.data.dispatch( 'core/notices' ).createNotice( 'error', 'Test 1', { id: 'test1' } );
wp.data.dispatch( 'core/notices' ).createNotice( 'error', 'Test 2', { id: 'test2' } );
wp.data.dispatch( 'core/notices' ).createNotice( 'error', 'Test 3', { id: 'test3' } );
wp.data.dispatch( 'core/notices' ).removeNotices( [ 'test1', 'test2', 'test3' ] );
wp.data.select( 'core/notices' ).getNotices();
```
3. Expect the select to return an empty array.
4. Run the following code:
```js
wp.data.dispatch( 'core/notices' ).createNotice( 'error', 'Test 1', { context: 'foo', id: 'foo1' } );
wp.data.dispatch( 'core/notices' ).createNotice( 'error', 'Test 2', { context: 'bar', id: 'bar1' } );
wp.data.dispatch( 'core/notices' ).removeNotices( [ 'foo1', 'bar1' ], 'foo' );
wp.data.select( 'core/notices' ).getNotices( 'bar' );
```
5. Expect the select to return an array containing the `Test 2` error.
